### PR TITLE
[weather][refactor] kill g_weatherManager global

### PR DESCRIPTION
**Pull request by xhaggi**
see title

### CODE REVIEW

**Review by notspiff — APPROVED** [review]
nothing spotted, nice cleanup

**Comment by xhaggi** [comment]
Ok thanks

**Comment by scott967** [comment]
I think this has caused a problem in reading locale / regional settings.  Changing temp unit or speed unit from default causes crash when loading giusettings on start.  See Trac #17706

**Comment by MilhouseVH** [comment]
@scott967 see #13319. Should be fixed in the next nightly.

File: xbmc/settings/Settings.cpp [hunk 1]
```diff
@@ -1012,7 +1012,7 @@ void CSettings::InitializeISettingCallbacks()
   settingSet.clear();
   settingSet.insert(CSettings::SETTING_WEATHER_ADDON);
   settingSet.insert(CSettings::SETTING_WEATHER_ADDONSETTINGS);
-  GetSettingsManager()->RegisterCallback(&g_weatherManager, settingSet);
+  GetSettingsManager()->RegisterCallback(&CServiceBroker::GetWeatherManager(), settingSet);
```
Comment (FernetMenta) [inline xbmc/settings/Settings.cpp]: This is not correct. @Montellese this mechanism for registering callbacks is based on globals and has become an issue now. Here it registers a method that does not exist at this time.
We need a mechanism where services can register dynamically.

Comment (xhaggi) [reply xbmc/settings/Settings.cpp]: I didn't think of it while refactored the weather manager. We are able to call `RegisterCallback` from `CWeatherManager`. There is no need to leave it here.

Comment (FernetMenta) [reply xbmc/settings/Settings.cpp]: Nice, I didn't know. Are you taking care of it?

Comment (xhaggi) [reply xbmc/settings/Settings.cpp]: see https://github.com/xbmc/xbmc/pull/13326

**Pull request merged**